### ISSUE TITLE
"Data block size X is not a multiple of 512" when creating null bdev with protection information enabled

### ISSUE THREAD

**Issue by karlatec** [spdk/spdk#1577]
## Current Behavior

Creating null bdev with "--md-size" and "--dif-type" options fail due to "Data block size 4096 is not a multiple of 512" error

## Steps to Reproduce

1.  sudo ./build/bin/vhost
2.  sudo scripts/rpc.py bdev_null_create Null0 100 4096 -t 1 -m 128
```
request:
{
  "name": "Null0",
  "num_blocks": 25600,
  "block_size": 4096,
  "md_size": 128,
  "dif_type": 1,
  "dif_is_head_of_md": true,
  "method": "bdev_null_create",
  "req_id": 1
}
Got JSON-RPC error response
response:
{
  "code": -22,
  "message": "Data block size 4096 is not a multiple of 512"
}
```

## Possible solution / question
But this works (although with an error): 
1. sudo scripts/rpc.py bdev_null_create Null0 100 4224 -t 1 -m 128
```
[2020-08-27 08:28:10.161091] vbdev_gpt.c: 552:vbdev_gpt_examine: *ERROR*: GPT module does not support block size 4224 for bdev Null0
Null0
```

2. sudo scripts/rpc.py bdev_get_bdevs
```
[
  {
    "name": "Null0",
    "aliases": [],
    "product_name": "Null disk",
    "block_size": 4224,
    "num_blocks": 24824,
    "uuid": "294e677a-3ec9-46e6-88fb-00dd0c8ee3cb",
    "md_size": 128,
    "md_interleave": true,
    "dif_type": 1,
    "dif_is_head_of_md": false,
    "enabled_dif_check_types": {
      "reftag": true,
      "apptag": false,
      "guard": true
    },
    "assigned_rate_limits": {
      "rw_ios_per_sec": 0,
      "rw_mbytes_per_sec": 0,
      "r_mbytes_per_sec": 0,
      "w_mbytes_per_sec": 0
    },
    "claimed": false,
    "zoned": false,
    "supported_io_types": {
      "read": true,
      "write": true,
      "unmap": false,
      "write_zeroes": true,
      "flush": false,
      "reset": true,
      "nvme_admin": false,
      "nvme_io": false
    },
    "driver_specific": {}
  }
]
```

This actually matches the behavior from bdev_null_rpc.c:
```
	data_block_size = req.block_size - req.md_size;
	if (data_block_size % 512 != 0) {
		spdk_jsonrpc_send_error_response_fmt(request, -EINVAL,
						     "Data block size %u is not a multiple of 512", req.block_size);
		goto cleanup;
	}
```
I couldn't find an answer anywhere in our docs, but is it the way it's supposed to be? Shouldn't we take 4096 as block_size argument and later increase it by md_size value, instead of making it a "user-side" operation when using rpc.py?

If that's as expected then we can probably close this as invalid issue, but I feel that then at least vbdev_gpt_examine error should be fixed, and relation between `block_size` and `md_size` better documented.

## Context (Environment including OS version, SPDK version, etc.)
SPDK b5fd1f8
Fedora 32 5.7.8-200.fc32.x86_64
gcc version 10.1.1

**Comment by AlekseyMarchuk** [comment]
When you use metadata, `block_size` is represented as extended LBA format, so it includes data size + metadata size. I think we should update our documentation

**Comment by karlatec** [comment]
Thanks @AlekseyMarchuk 
If this is as expected, can we also fix the GPT error besides documentation?

**Comment by shuhei-matsumoto** [comment]
Hi @karlatec 

It is known that SPDK GPT doesn’t support metadata. I tried to support quickly before but failed. For GPT will you add an new issue as enhancements?

**Comment by karlatec** [comment]
Sure thing

**Comment by AlekseyMarchuk** [comment]
@karlatec has updated the description of `block_size` parameter in https://review.spdk.io/gerrit/c/spdk/spdk/+/3949, so this patch also fixes this problem. I'm assigning this issue to Karol, it can be closed once the patch is merged

**Comment by karlatec** [comment]
Doc fixed by https://review.spdk.io/gerrit/c/spdk/spdk/+/3949
GPT error part will be resolved in #1580
